src/components/Calendar/indicators.js

```diff
@@ -1,38 +1,59 @@
-import React, { PropTypes } from 'react';
+import React, { PropTypes, Component } from 'react';
 import { compose } from '../../utils/utils';
 import Day from '../Day/Day';
 import { indicatorDay } from '../Day/indicatorDay';
 import { isSameDay } from '../../utils/date-utils';
 
 /**
- * Higher Order Component to add "day indicators" feature
- * @param {Component} Component
+ * Higher Order Component to add "day indicators" feature to Calendar
+ * @param {Component} CalendarComponent
  * @return {IndicatorCalendar}
  */
-export function indicators(Component) {
-  class IndicatorCalendar extends React.Component {
+export function indicators(CalendarComponent) {
+  class IndicatorCalendar extends Component {
     constructor() {
       super(...arguments);
 
+      this.state = {
+        // Create enhancedDayComponent and store in state.
+        enhancedDayComponent: this.enhanceDayComponent(this.props.DayComponent),
+      };
+    }
+
+    componentDidUpdate(prevProps) {
+      // We don't want to create this on every render, but we also want
+      // to account for the fact that props.DayComponent could change
+      // We only want to re-create if props.DayComponent changed.
+      // Treat this as a "computed property".
+      if (prevProps.DayComponent !== this.props.DayComponent) {
+        const enhancedDayComponent = this.enhanceDayComponent(this.props.DayComponent);
+        this.setState({enhancedDayComponent});
+      }
+    }
+
+    /**
+     * @param {Component} DayComponent
+     * @return {Component} The enhanced DayComponent
+     */
+    enhanceDayComponent(DayComponent) {
       const hasIndicator = (date) => this.props.events.some(
         (event) => isSameDay(event.date, date)
       );
 
-      // The enhanced props.DayComponent
-      this.DayComponent = compose([
```

Comment:
The main idea [in this commit](https://github.com/erikthedeveloper/react-calendar-components/pull/2/commits/7643bba6db52996ed65c1e64f7fb7cc3c5637839?diff=split) is to avoid creating and setting this on `this.DayComponent`. 

While this works for this use case and doesn't break anything, this is a relatively terrible habit. This approach falls apart if `props.DayComponent` changes (more specifically if `props.*` changes, anything we derived from the previous props is now stale since we naively derived it only once in the initial `constructor`). 

Summary: Avoid teaching bad habits and focus more on the approach rather than this specific use "building this example Calendar" use case.